packages/@aws-cdk/aws-pipes-targets-alpha/test/integ.api-destination.ts

```diff
@@ -0,0 +1,104 @@
+
+/*
+ * This integration test creates an API Gateway HTTP API in order to
+ * validate that the pipe executes properly. The flow is as follows:
+ *
+ * SQS (pipe source) --> EventBridge API destination (pipe target)
+ * --> API Gateway HTTP API --> Lambda function
+ */
+
+class TestSource implements ISource {
+  sourceArn: string;
+  sourceParameters = undefined;
+  constructor(private readonly queue: cdk.aws_sqs.Queue) {
+    this.queue = queue;
+    this.sourceArn = queue.queueArn;
+  }
+  bind(_pipe: IPipe): SourceConfig {
+    return {
+      sourceParameters: this.sourceParameters,
+    };
+  }
+  grantRead(pipeRole: cdk.aws_iam.IRole): void {
+    this.queue.grantConsumeMessages(pipeRole);
+  }
+}
+
+const fn = new lambda.Function(stack, 'ConnectHandler', {
+  runtime: lambda.Runtime.NODEJS_LATEST,
+  handler: 'index.handler',
+  code: new lambda.InlineCode('exports.handler = async function(event, context) { console.log(event); return { statusCode: 200, body: "connected" }; };'),
+});
+
+const integration = new HttpLambdaIntegration('LambdaIntegration', fn);
+const httpApi = new apigwv2.HttpApi(stack, 'HttpApi');
+httpApi.addRoutes({
+  path: '/books',
+  methods: [apigwv2.HttpMethod.GET],
+  integration,
+});
+
+const secret = new cdk.aws_secretsmanager.Secret(stack, 'MySecret', {
+  secretStringValue: cdk.SecretValue.unsafePlainText('abc123'),
+});
+
+const connection = new cdk.aws_events.Connection(stack, 'MyConnection', {
+  authorization: cdk.aws_events.Authorization.apiKey('x-api-key', secret.secretValue),
+  description: 'Connection with API Key x-api-key',
+  connectionName: 'MyConnection',
+});
+
+const destination = new cdk.aws_events.ApiDestination(stack, 'MyDestination', {
+  connection,
+  endpoint: `${httpApi.apiEndpoint}/books`,
+  httpMethod: cdk.aws_events.HttpMethod.GET,
+  apiDestinationName: 'ApiGwHttpApi',
+  rateLimitPerSecond: 1,
+  description: 'Calling API GW HTTP API',
+});
+
+new Pipe(stack, 'Pipe', {
+  source: new TestSource(sourceQueue),
+  target: new ApiDestinationTarget(destination, {
+    headerParameters: {
+      'x-header': 'myheader',
+      'x-api-key': 'apiKeyFromHeaderParams', // check which header has precedence
+    },
+    queryStringParameters: { key: 'harrypotter' },
+  }),
+});
+
+const test = new IntegTest(app, 'integtest-pipe-target-api-dest', {
+  testCases: [stack],
+});
+
+const putMessageOnQueue = test.assertions.awsApiCall('SQS', 'sendMessage', {
+  QueueUrl: sourceQueue.queueUrl,
+  MessageBody: 'USA',
+});
+
+const logEvents = test.assertions.awsApiCall('CloudWatchLogs', 'filterLogEvents', {
+  logGroupName: fn.logGroup.logGroupName,
+  limit: 1,
+});
+
+const message = putMessageOnQueue.next(logEvents);
+
+// Check that the Lambda was invoked successfully from API GW
+message.assertAtPath('events.0.message', ExpectedResult.stringLikeRegexp('INIT_START')).waitForAssertions({
+  totalTimeout: cdk.Duration.minutes(1),
+  interval: cdk.Duration.seconds(15),
+});
```

Comment:
Correct me if I'm wrong, but I'm not sure whether this check is explicit enough to verify that the Lambda was invoked. Since the code is throwing a 200 return code, could we check for that instead? Or can we verify that the SQS message was sent by checking for the message body?

Reply:
I updated the integration test to check for the payload from SQS, which is sent as follows:

`queryStringParameters: { key: '$.body' },`